packages/react/src/test/appendBody.ts

```diff
@@ -0,0 +1,46 @@
+import { IncomingMessage, ServerResponse } from 'http';
+import {
+    ExpressSynthqlHandler,
+    ExpressSynthqlHandlerRequest,
+    ExpressSynthqlHandlerResponse,
+} from '@synthql/backend';
+
+function readBody(req: IncomingMessage): Promise<string> {
+    return new Promise((resolve, reject) => {
+        let body = '';
+
+        req.on('data', (chunk) => {
+            body += chunk;
+        });
+
+        req.on('end', () => {
+            resolve(body);
+        });
+
+        req.on('error', (e) => {
+            reject(e);
+        });
+    });
+}
+
+export type IncomingMessageWithBody = IncomingMessage &
+    ExpressSynthqlHandlerRequest;
+
+export type ServerResponseWithEnd = ServerResponse &
+    ExpressSynthqlHandlerResponse;
+
+export function appendBody(expressHandler: ExpressSynthqlHandler) {
+    return async (req: IncomingMessage, res: ServerResponse) => {
+        const body = await readBody(req);
+
+        const newReq = {
+            ...req,
+            headers: req.headers,
+            body: body,
+        } as IncomingMessageWithBody;
```

Comment:
don't do `as Type`, this is generally unsafe behaviour. Instead do

```ts
 const newReq: Type = { ... }
```

Reply:
This would be ideal, but the goal here is actually type casting. We don't have all the values for the type, so using a type annotation in this case, causes a type mismatch error.